var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -368,6 +368,21 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
 
     # ###################### Dependencies ##########################
 
+    # External Kokkos
+    depends_on("kokkos@4.1.00", when="@14.4.0: +kokkos")
```

Comment:
Should this be so exact? Perhaps
```suggestion
    depends_on("kokkos@4.1:4", when="@14.4.0: +kokkos")  # Trilinos 14.4.0 vendors Kokkos 4.1.00
```

Reply:
Hi @sethrj. This is just my opinion, but I do think this needs to be this exact, at least for now. This was a large concern with allowing Trilinos to build against an external Kokkos, namely that Trilinos developers are (have historically been) creating code against a particular vendored version of Kokkos.

I believe that Trilinos CI testing is also only being performed on the vendored version. Perhaps in the future, additional Trilinos CI testing could be added against some minimal version \# for a major release of Kokkos (like 4.0.0) to test for backwards compatibility against a major version (all 4.x.x).

Reply:
Fair enough. If Kokkos is doing version compatibility right, however, it should allow a more generous version range. I could imagine this being desirable if Kokkos gets a patch release and can deploy at a faster rate than Trilinos.